docs/fp_primer.md

```diff
@@ -487,30 +524,288 @@ This algorithmically looks something like this:
+
+Compile time evaluation is execution that occurs when setting up a shader
+module, i.e. when compiling WGSL to a platform specific shading language. It is
+part of resolving values for things like constants, and occurs once before the
+shader is run by the caller. It includes constant evaluation and override
+evaluation. All AbstractFloat operations are compile time evaluated.
+
+Runtime evaluation is execution that occurs every time the shader is run, and
+may include dynamic data that is provided between invocations. It is work that
+is sent to the GPU for execution in the shader.
+
+### Behavioural Differences
+
+For a well-defined operation with a finite response, runtime and compile time
+evaluation should be indistinguishable.
+
+For example:
+```
+// runtime
+@group(0) @binding(0) var a : f32;
+@group(0) @binding(1) var b : f32;
+
+let c: f32 = a + b
+```
+and
+```
+// compile time
+const c: f32 = 1 + 2
+```
+should produce the same result of `3` in the variable `c`, assuming `1` and `2`
+were passed in as `a` & `b`.
+
+The only difference, is when/where the execution occurs.
+
+The difference in behaviour between these two occur when the result of the
+operation is not finite for the underlying floating point type.
+
+If instead of `1` and `2`, we had `10` and `f32.max`, so the true result is
+`f32.max + 10`, the user will see a difference. Specifically the runtime
+evaluated version will still run, but the result in `c` will be an indeterminate
+value, which is any finite f32 value. For the compile time example instead,
+compiling the shaderwill fail validation.
+
+This applies to any operation, and isn't restricted to just addition. Anytime a
+value goes outside the finite values the shader will hit these results,
+indeterminate for runtime execution and validation failure for compile time
+execution.
+
+Unfortunately we are dealing with intervals of results and not precise results.
+So this leads to more even conceptual complexity. For runtime evaluation, this
+isn't too bad, because the rule becomes if any part of the interval is
+non-finite then an indeterminate value can be a result, and the interval for an
+indeterminate result `[min, max]`, will include any finite portions of the
+interval.
+
+Compile time evaluation becomes significantly more complex, because difference
+isn't what interval is returned, but does this shader compile or not, which are
+mutually exclusive. This is compounded even further by having to consider out of
+bounds vs overflow behaviour. Thankfully this can be broken down into a case by
+case basis based on where an interval falls.
+
+Assuming `X`, is the well-defined result of an operation, i.e. not an
+indeterminate result because the operation isn't defined for the inputs:
+
+| Region                 |                                                  | Result                         |
+|------------------------|--------------------------------------------------|--------------------------------|
```

Comment:
Here you can repeat with terms:   finite, far-overflow, near-overflow

Reply:
I am not 100% sure what you are suggesting here. Are you wanting the definitions of these terms to be repeated here?

Reply:
You already did it. You used the far-overflow and near-overflow in the description column.